services/api/service.go

```diff
@@ -316,6 +331,8 @@ func (api *RelayAPI) StartServer() (err error) {
 		for {
 			headEvent := <-c
 			api.processNewSlot(headEvent.Slot)
+			// TODO(mikeneuder): Add checks to make sure we haven't
+			// optimistically built on top of a non-simulated slot.
```

Comment:
What does this `TODO` mean?

Reply:
this was something you mentioned briefly where we don't want to chain optimistic slots together. i understood it as wanting to avoid processing blocks optimistically before the previous slot block has been successfully simulated. but i wasn't sure if i understood correctly or how to incorporate it so i left it as a TODO. WDYT?

Reply:
> avoid processing blocks optimistically before the previous slot block has been successfully simulated

What I meant to say is: don't give an optimistic builder the possibility of cheating in >1 slots (because otherwise the builder's collateral wouldn't cover the liability). If we are still simulating blocks for a given optimistic builder at slot `n`, pessimistically simulate blocks at slots `>n` until simulation for slot `n` confirms the block was good.

Reply:
oh i see. got it. ya it sounds like we will need another field in redis that stores this so we can check on the fast path. let me think on this a bit more and circle back to this comment with a plan.

Reply:
i think i found a pretty elegant way to handle this! https://github.com/michaelneuder/mev-boost-relay/pull/1/commits/a2c96626377ad726eb25ef3b76143b53fa6198d6

basically, we only need to enforce this condition if the builder proposer the winning block for slot N. So in the `handleGetPayload` function, while we process the winning block, we can just manually set the builder down to high-prio rather than optimistic. as long as the simulation succeeds, we will reset them back to optimistic.

Reply:
Yep, that makes a lot of sense to me :) Nice!

Reply:
:-)